decompressed/base/lib/mount_modroot/05_transfer_basefiles

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+
+upperdir=$1
+overlay_bank=/overlay/bank_2
+
+move_overlay() {
+  #use the config folder as check to know that we should populate the modoverlay (mtd3) and cleanup real overlay (mtd2)
+  if [ -d $overlay_bank/etc/config ]; then
+
+    #copy all file from overlay_bank to modoverlay keeping all attributes and links
+    cp -a $overlay_bank/* $upperdir/
+
+    #at this point we should remove all empty folders in bank_2 overlay, except the modoverlay mountpoint
+    for f in "$overlay_bank/"*; do
+      [ "$f" != "$overlay_bank$upperdir" ] && rm -rf "$f"
```

Comment:
qua c'è un problema, il mountpoint da non rimuovere non è $overlay_bank$upperdir ma $overlay_bank/modoverlay

/overlay/bank_2/modoverlay/bank_mod non esiste in nessun caso, quindi questo check non si attiva mai e il mountpoint viene eliminato

Reply:
giusto ma questo cosa può causare a livello di script?

Reply:
bho, è quello che dicevamo tempo fa con @FrancYescO 

il comando di sopra copia la cartella vuota in mtd3, ma la cartella sulla quale mtd3 è montata, che dovrebbe essere qui in mtd2 sparisce.

In linux non puoi eliminare un mountpoint mentre c'è montato qualcosa dentro, ma in overlayfs puoi eliminare un elemento direttamente dall'upper layer senza eseguire l'eliminazione sull'overlay. Quindi immagino succedano cose random laddove dopo l'rm ci sia ancora in sospeso qualche operazione in scrittura.

Reply:
Inoltre il fatto che poco sopra la cartella /overlay/bank_2/modoverlay sia stata copiata dentro /modoverlay indi per cui dopo il pivot /modoverlay (dove è montata mtd3) sia in realtà salvata proprio dentro mtd3 non aiuta

Reply:
aggiungerei agli skip anche saferoot